iggy/src/users/delete_user.rs

```diff
@@ -0,0 +1,100 @@
+use crate::bytes_serializable::BytesSerializable;
+use crate::command::CommandPayload;
+use crate::error::Error;
+use crate::identifier::Identifier;
+use crate::validatable::Validatable;
+use serde::{Deserialize, Serialize};
+use std::fmt::Display;
+use std::str::FromStr;
+
+#[derive(Debug, Serialize, Deserialize, PartialEq, Default)]
+pub struct DeleteUser {
+    #[serde(skip)]
+    pub user_id: Identifier,
+}
+
+impl CommandPayload for DeleteUser {}
+
+impl Validatable<Error> for DeleteUser {
+    fn validate(&self) -> Result<(), Error> {
+        Ok(())
+    }
+}
+
+impl FromStr for DeleteUser {
+    type Err = Error;
+    fn from_str(input: &str) -> Result<Self, Self::Err> {
+        let parts = input.split('|').collect::<Vec<&str>>();
+        if parts.len() != 1 {
+            return Err(Error::InvalidCommand);
+        }
+
+        let stream_id = parts[0].parse::<Identifier>()?;
+        let command = DeleteUser { user_id: stream_id };
+        command.validate()?;
+        Ok(command)
+    }
+}
+
+impl BytesSerializable for DeleteUser {
+    fn as_bytes(&self) -> Vec<u8> {
+        self.user_id.as_bytes()
+    }
+
+    fn from_bytes(bytes: &[u8]) -> Result<DeleteUser, Error> {
+        if bytes.len() < 3 {
+            return Err(Error::InvalidCommand);
+        }
+
+        let stream_id = Identifier::from_bytes(bytes)?;
+        let command = DeleteUser { user_id: stream_id };
+        command.validate()?;
+        Ok(command)
+    }
+}
+
+impl Display for DeleteUser {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.user_id)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn should_be_serialized_as_bytes() {
+        let command = DeleteUser {
+            user_id: Identifier::numeric(1).unwrap(),
+        };
+
+        let bytes = command.as_bytes();
+        let stream_id = Identifier::from_bytes(&bytes).unwrap();
```

Comment:
stream_id is a bit confusing - I'd use jus user_id

Reply:
Correct, good catch, will update.